frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/DivergedFromRemoteDialog.kt

```diff
@@ -0,0 +1,110 @@
+package com.virtuslab.gitmachete.frontend.actions.dialogs
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.ui.dsl.builder.MutableProperty
+import com.intellij.ui.dsl.builder.panel
+import com.virtuslab.gitmachete.backend.api.IManagedBranchSnapshot
+import com.virtuslab.gitmachete.backend.api.IRemoteTrackingBranchReference
+import com.virtuslab.gitmachete.backend.api.SyncToRemoteStatus
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.format
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString
+
+enum class DivergeResolutionOption {
+  FORCE_PUSH,
+  RESET_ON_REMOTE,
+  MERGE_REMOTE_INTO_LOCAL,
+  REBASE_ON_REMOTE
+}
+
+class DivergedFromRemoteDialog(
+  project: Project,
+  private val remoteBranch: IRemoteTrackingBranchReference,
+  private val branch: IManagedBranchSnapshot,
+  private val relationToRemote: SyncToRemoteStatus
+) : DialogWrapper(project, /* canBeParent */ true) {
+
+  private var divergeResolutionOption = DivergeResolutionOption.REBASE_ON_REMOTE
+
+  init {
+    title =
+      getString("action.GitMachete.TraverseRepositoryAction.dialog.diverged-from-remote.title")
+    setOKButtonMnemonic('O'.code)
+    super.init()
+  }
+
+  val relationToRemoteDescription =
+    when (relationToRemote) {
+      SyncToRemoteStatus.DivergedFromAndNewerThanRemote ->
+        getString("action.GitMachete.TraverseRepositoryAction.dialog.diverged-from-remote.newer-than-remote")
+      SyncToRemoteStatus.DivergedFromAndOlderThanRemote ->
+        getString("action.GitMachete.TraverseRepositoryAction.dialog.diverged-from-remote.older-than-remote")
+      else -> { "" }
+    }
+
+  fun showAndGetThePreferredAction() =
+    if (showAndGet()) {
+      divergeResolutionOption
+    } else {
+      null
+    }
+
+  override fun createCenterPanel() = panel {
+    row {
+      label(
+        format(
+          getString(
+            "action.GitMachete.TraverseRepositoryAction.dialog.diverged-from-remote.text.HTML"
+          ),
+          branch.name,
+          remoteBranch.fullName,
+          relationToRemoteDescription
+        )
+      )
+    }
+    buttonsGroup {
+      row {
```

Comment:
Options to be suggested based on whether the branch is older or newer than its remote:
* if local older than remote, then suggest reset to remote.
* if local newer than remote, then suggest force push or (questionable but okay) rebase onto remote

Reply:
when diverge from remote?

Reply:
> when diverge from remote?

🤔 could you elaborate? ofc both cases (_if local older than remote_ and _if local newer than remote_ already refer to the case of local branch diverging from remote)

Reply:
well, when committing some reviews on the remote, and completing some changes you were working on locally and committing them after that. Now you want to keep both.

Reply:
Okay... so in terms of _local newer/older then remote_, I think if matches both cases.
So the update break-down of cases would be:
* if local older than remote, then suggest reset to remote **or** rebase onto remote
* if local newer than remote, then suggest force push  **or** rebase onto remote